AsyncDisplayKit/ASDisplayNode.h

```diff
@@ -506,7 +506,18 @@ typedef CALayer *(^ASDisplayNodeLayerBlock)();
 @interface ASDisplayNode (UIViewBridge)
 
 - (void)setNeedsDisplay;    // Marks the view as needing display. Convenience for use whether view is created or not, or from a background thread.
-- (void)setNeedsLayout;     // Marks the view as needing layout.  Convenience for use whether view is created or not, or from a background thread.
+
+/**
+ * Marks the view as needing layout. Convenience for use whether view is created or not, or from a background thread.
+ * 
+ * If this node was measured, calling this method triggers an internal relayout: the calculated layout is invalidated,
+ * and the supernode is notified or (if this node is the root one) a full measurement pass is executed using the old constrained size.
+ * 
+ * Note: If the relayout causes a change in size of the root node that is attached to a container view
+ * (table or collection view, for example), the container view must be notified to redraw. 
```

Comment:
Yes, as I already mentioned in my notes, a delegate is best fit for this case. It will facilitate an automatic and flexible relayout model. I haven't implemented it yet and will create a follow-up ticket once this PR is merged. I'm not even sure about how the new API should be, so please take on that work if you have time.

Very good points on the comment. I will update it very soon. 1 tidbit though, I'm gonna write "an empty batch editing transaction **must** be triggered" instead of "will", because without the delegate we still need this to be done in client code.